Automattic/vip-go-mu-plugins
# Add the global timeout for search queries

## Description

Refactor timeouts for client and backend to use values globally defined as class constants rather than hardcoded values.

Utilize `ep_post_formatted_args` filter to set the timeout on ES post search queries.

## Changelog Description

### Plugin Updated: Enterprise Search

We've implemented a timeout on Elasticsearch queries, it matches the Enterprise Search client timeout so there's not going to be a difference in terms of behavior. It's meant as a safety mechanism to prevent runaway queries after the client has already disconnected.


## Checklist

Please make sure the items below have been covered before requesting a review:

- [] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Check out PR.
2. Do a search query
1. Verify in Search Dev Tools that payload has `timeout` in it.
